R-package/R/lgb.Dataset.R

```diff
@@ -45,28 +49,92 @@ Dataset <- R6::R6Class(
           stop("lgb.Dataset: If provided, predictor must be a ", sQuote("lgb.Predictor"))
       }
 
+      # Create known attributes list
+      if (!is.null(label)) info[["label"]] <- label
+      if (!is.null(weight)) info[["weight"]] <- weight
+      if (!is.null(init_score)) info[["init_score"]] <- init_score
+      if (!is.null(group)) info[["group"]] <- group
```

Comment:
```suggestion
      if (!is.null(label)) {
        info[["label"]] <- label
      }
      if (!is.null(weight)) {
        info[["weight"]] <- weight
      }
      if (!is.null(init_score)) {
        info[["init_score"]] <- init_score
      }
      if (!is.null(group)) {
        info[["group"]] <- group
      }
```

Following the style in the rest of the R package, please use `{}` for the expression in every control statement (`for`, `if`, `while`) and do not perform assignments in one-liners.